ledger/simulation/tracer.go

```diff
@@ -302,12 +309,28 @@ func (o *OpcodeTraceUnit) appendAddedStackValue(cx *logic.EvalContext, tracer *e
 	}
 }
 
+func (tracer *evalTracer) recordCurrentScratchChange(cx *logic.EvalContext) {
+	scratchSlotID, newValue, isStoreAlike := cx.CurrentScratchChange()
```

Comment:
Just a nit: it would be nice if this code (which is executed in BeforeOpcode) only reported the scratch slot ID which changed, then in AfterOpcode you look at the actual scratch slot data structure to pull out the new value.

This doesn't matter much, but it would close the gap between what we report and what the evaluator actually does. And I imagine for reporting more complicated state in the future (i.e. boxes), you can't just rely on the stack value, so establishing this pattern early can't hurt.

Reply:
fair point, see [pattern build for b4/after opcode](https://github.com/algorand/go-algorand/pull/5563/commits/272c206196960c0d830901b8046710470ec3e36d)